src/main/java/com/rultor/agents/IndexesRequests.java

```diff
@@ -39,8 +45,103 @@
  * @version $Id$
  */
 public final class IndexesRequests implements SuperAgent {
+    /**
+     * Xpath expression for retrieving log entries from a talk's XML.
+     */
+    private static final String ARCHIVE_LOG = "//archive/log";
+
+    /**
+     * Name of the log index attribute.
+     */
+    private static final String INDEX = "index";
+
     @Override
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
     public void execute(final Talks talks) throws IOException {
-        throw new UnsupportedOperationException();
+        int maxTalk = this.getMaxTalkIndex(talks);
+        for (final Talk talk : talks.active()) {
+            final List<String> requests = talk.read().xpath("//request");
+            if (requests.isEmpty()) {
+                int index = 0;
+                final List<XML> logs = talk.read().nodes(ARCHIVE_LOG);
+                if (logs.isEmpty()) {
+                    index = maxTalk + 1;
+                } else {
+                    index = this.getMaxLogIndex(logs) + 1;
+                }
+                talk.modify(
+                    new Directives().xpath("//talk").add("request")
+                        .attr(INDEX, Integer.toString(index))
+                        .attr("id", this.createRequestId())
+                        .add("type").set(INDEX)
+                        .up()
+                        .add("args")
+                );
+                maxTalk += 1;
+            }
+        }
+    }
+
+    /**
+     * Returns the greatest index of all log children of an archive node.
+     * @param logs A list of XML objects for individual log nodes (e. g.
+     *  <log id="3" index="1" title="title3"/>
+     * @return Highest value of the index attribute of all log nodes contained
+     *  in the logs list.
+     */
+    private int getMaxLogIndex(final List<XML> logs) {
+        int max = 0;
+        for (final XML log : logs) {
+            int index = 0;
+            final List<String> texts = log.xpath("@index");
+            if (texts.size() == 1) {
+                final String text = texts.get(0);
+                try {
+                    index = Integer.parseInt(text);
+                } catch (final NumberFormatException exception) {
+                    Logger.error(
+                        this,
+                        String.format(
+                            "Invalid index number '%s'",
+                            text
+                        )
+                    );
```

Comment:
What this is for? Don't try to save the software, make it as fragile as possible. Let it break and fail! Read about fail fast: http://www.martinfowler.com/ieeeSoftware/failFast.pdf